api/task_processor/models.py

```diff
@@ -0,0 +1,74 @@
+import pickle
+import typing
+import uuid
+from datetime import datetime
+
+from django.db import models
+from django.utils import timezone
+
+
+class Task(models.Model):
+    uuid = models.UUIDField(unique=True, default=uuid.uuid4)
+    created_at = models.DateTimeField(auto_now_add=True)
+    scheduled_for = models.DateTimeField(blank=True, null=True, default=timezone.now)
+
+    pickled_callable = models.BinaryField()
```

Comment:
Yeah, I was a little concerned about the use of pickle (hence why I tried to extract it as much as possible to make it easier to change). I took a long time deciding between json and pickle but went with pickle because of the JSON serialisation issues - your points about pickle are well founded though so I guess those issues are the path of least resistance. 

And yes, the decorator stuff was my next task (excuse the pun). Like you've suggested, I'd like to get it as close to celery as possible so we don't have to do much to replace it in the future if we have to.